aspnet/fundamentals/file-providers.rst

```diff
@@ -1,8 +1,149 @@
-.. include:: /../common/stub-topic.txt
+:version: 1.0.0
 
-|stub-icon| File Providers
-==========================
+File Providers
+==============
 
-.. include:: /../common/stub-notice.txt
+By `Steve Smith`_
+
+ASP.NET Core abstracts file system access through the use of File Providers.
+
+.. contents:: Sections:
+  :local:
+  :depth: 1
+
+`View or download sample code <https://github.com/aspnet/Docs/tree/master/aspnet/fundamentals/file-providers/sample>`__
+
+File Provider Abstractions
+--------------------------
+
+File Providers are an abstraction over file systems. The main interface is :dn:iface:`~Microsoft.Extensions.FileProviders.IFileProvider`. ``IFileProvider`` exposes methods to get file information (:dn:iface:`~Microsoft.Extensions.FileProviders.IFileInfo`), directory information (:dn:iface:`~Microsoft.Extensions.FileProviders.IDirectoryContents`), and to set up change notifications (using an :dn:iface:`~Microsoft.Extensions.Primitives.IChangeToken`). These types wrap the ``System.IO.File`` type, scoping all paths to a directory and its children. This scoping limits access to a certain directory and its children, preventing access to the file system outside of this boundary.
```

Comment:
The last part, starting with "These types wrap" is only true for the physical file provider. Embedded doesn't have directories. It wraps embedded resources and uses namespaces instead of directories.